### ISSUE TITLE
Regex checks causing `java.lang.ArrayIndexOutOfBoundsException`

### ISSUE THREAD

**Issue by heshanpadmasiri** [ballerina-platform/nballerina#1073]
Regex pair

- `WR(eN)*U(bZn)*(kPC(xeq(KdVlPh)*AWfCKC)*oePqmj)*`
- `WR(eN)*U(bZn)*(kPC(xeq(KdVl(VeRKOGFRr)*Ph)*AWfCKC)*oePqmj)*`

causes fallowing error:
```
WARNING: Incompatible JRE version '17.0.4' found. This ballerina program supports running on JRE version '11.0.*'
[2022-09-08 20:10:54,868] SEVERE {b7a.log.crash} - Index 256 out of bounds for length 256
java.lang.ArrayIndexOutOfBoundsException: Index 256 out of bounds for length 256
        at wso2.nballerina$0046types.0.list.listSubtypeIsEmpty(list.bal:118)
        at wso2.nballerina$0046types.0.nballerina$$$types.listSubtypeIsEmpty$lambda39$(nballerina.types:562)
        at wso2.nballerina$0046types.0.core.isEmpty(core.bal:778)
        at wso2.nballerina$0046types.0.core.isSubtype(core.bal:787)
        at wso2.nballerina$0046types$0046regex.0.regex.typeRelation(regex.bal:30)
        at wso2.nballerina.0.main.regexTypeRelation(main.bal:121)
        at wso2.nballerina$0046regex.0.main.main(main.bal:8)
        at wso2.nballerina$0046regex.0.$_init.$lambda$main$(nballerina.regex)
        at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(Scheduler.java:611)
        at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:317)
        at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:284)
        at java.base/java.lang.Thread.run(Thread.java:833)

[2022-09-08 20:10:54,889] SEVERE {b7a.log.crash} - Index 256 out of bounds for length 256
java.lang.ArrayIndexOutOfBoundsException: Index 256 out of bounds for length 256
        at wso2.nballerina$0046types.0.list.listSubtypeIsEmpty(list.bal:118)
        at wso2.nballerina$0046types.0.nballerina$$$types.listSubtypeIsEmpty$lambda39$(nballerina.types:562)
        at wso2.nballerina$0046types.0.core.isEmpty(core.bal:778)
        at wso2.nballerina$0046types.0.core.isSubtype(core.bal:787)
        at wso2.nballerina$0046types$0046regex.0.regex.typeRelation(regex.bal:30)
        at wso2.nballerina.0.main.regexTypeRelation(main.bal:121)
        at wso2.nballerina$0046regex.0.main.main(main.bal:8)
        at wso2.nballerina$0046regex.0.$_init.$lambda$main$(nballerina.regex)
        at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(Scheduler.java:611)
        at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:317)
        at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:284)
        at java.base/java.lang.Thread.run(Thread.java:833)

[2022-09-08 20:10:54,890] SEVERE {b7a.log.crash} - Index 256 out of bounds for length 256
java.lang.ArrayIndexOutOfBoundsException: Index 256 out of bounds for length 256
        at wso2.nballerina$0046regex.0.main.main(main.bal:6)
        at wso2.nballerina$0046regex.0.$_init.$lambda$main$(nballerina.regex)
        at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(Scheduler.java:611)
        at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:317)
        at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:284)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

Based on our previous experience this is probably some sort of stack overflow in `listSubtypeIsEmpty`

In the older version (2eae9397791de8ca0fa05ef305eab40c2bbd9719) this produce incorrect answer as equal.

**Comment by jclark** [comment]
Not clear to me that the bug is in  nballerina. What is the evidence this is a stack overflow? jBallerina should be giving a more helpful error message in any case.

I would expect us to use quite a lot of stack space for complex types.

I suggest reporting this to jBallerina.

**Comment by heshanpadmasiri** [comment]
Closing because issue seems to have been fixed in the newer version of ballerina (2201.1.1)
